### PR TITLE
Setstate/initialState support removed

### LINKED ISSUE #1499: Remove initialState/setState
You can define state in each example using a custom initialState/setState API:

```jsx
initialState = { count: 42 }
;<Button onClick={() => setState({ count: state.count + 1 })}>
  {state.count}
</Button>
```

This was implemented long before hooks but does the same thing as hooks:

```jsx
const [count, setCount] = React.useState(42)
;<Button onClick={() => setCount(count + 1)}>{count}</Button>
```

Now the minimum supported by Styleguidist version of React supports hooks, so we can remove our custom API.

**This will be a breaking change.**

- [ ] Remove [code for initialState/setState](https://github.com/styleguidist/react-styleguidist/blob/ef4c109bf6b7dc594346332c9091fdab3e83c9b4/src/client/rsg-components/ReactExample/ReactExample.tsx#L28-L48)
- [ ] Update examples in the docs to hooks

### PR DESCRIPTION
**Pull request by thecodejack**
- removed support for setState and initialState
- Documentation cleanup

closes #1499

### CODE REVIEW

**Comment by thecodejack** [comment]
@sapegin as I mentioned in #1499 , we can merge https://github.com/styleguidist/react-styleguidist/commit/a0a80613538233dd3a264feecd4f5f7adc516b7e right away and remaining commits with major release.

File: docs/Cookbook.md [hunk 1]
```diff
@@ -452,18 +452,20 @@ Object references may not work as expected in examples state:
 
 ```jsx
```
Comment (sapegin) [inline docs/Cookbook.md]: This example is no longer relevant with hooks and we can remove the whole question.

File: docs/Documenting.md [hunk 2]
```diff
@@ -272,44 +272,35 @@ import Placeholder from 'rsg-example/components/Placeholder'
 
 > **Note:** You can only use `import` by editing your Markdown files, not by editing the example code in the browser.
 
-Each example has its own state that you can access as `state` variable and change with `setState()` function. Default state is `{}` and can be set with `initialState`.
+Each example acts as function component and you can use React Hook `useState` to handle its own state.
```
Comment (sapegin) [inline docs/Documenting.md]:
```suggestion
Each example acts as function component and you can use the `useState` Hook to handle its own state.
```

File: docs/Documenting.md [hunk 3]
```diff
@@ -272,44 +272,35 @@ import Placeholder from 'rsg-example/components/Placeholder'
...
 
 ````jsx
 // ```jsx inside Markdown
-initialState = { isOpen: false }
+const [isOpen, setisOpen] = React.useState(false)
```
Comment (sapegin) [inline docs/Documenting.md]:
```suggestion
const [isOpen, setIsOpen] = React.useState(false)
```

File: docs/Documenting.md [hunk 4]
```diff
@@ -272,44 +272,35 @@ import Placeholder from 'rsg-example/components/Placeholder'
...
+  <Modal isOpen={isOpen}>
     <h1>Hallo!</h1>
-    <button onClick={() => setState({ isOpen: false })}>Close</button>
+    <button onClick={() => setisOpen(false)}>Close</button>
   </Modal>
 </div>
 ````
 
-`initialState`, `state` and `setState()` helpers are good to show components in different states, but to let users copy-paste your example code without modifications into their React app you may want to use `React.Component` instead. We can rewrite the example above like this:
+To let users copy-paste your example code without modifications into their React app you may want to share full code instead. We can rewrite the example above like this:
```
Comment (sapegin) [inline docs/Documenting.md]: This is also no longer relevant and the example is now the same as the previous one. Examples with hooks are copy-pastable as is.

File: examples/sections/src/components/ThemeButton/Readme.md [hunk 5]
```diff
@@ -3,9 +3,9 @@ An example of using React Context with a component:
 ```jsx
 import ThemeContext from '../../ThemeContext.js'
 
-initialState = { theme: 'light' }
+const [theme, settheme] = React.useState('light')
```
Comment (sapegin) [inline examples/sections/src/components/ThemeButton/Readme.md]:
```suggestion
const [theme, setTheme] = React.useState('light')
```

File: src/client/rsg-components/ReactExample/ReactExample.tsx [hunk 6]
```diff
@@ -7,46 +7,6 @@ import splitExampleCode from '../../utils/splitExampleCode';
 
 /* eslint-disable react/no-multi-comp */
 
-// Wrap the example component with a Functional Component to support
-// hooks in examples
-function FunctionComponentWrapper(props: {
```
Comment (sapegin) [inline src/client/rsg-components/ReactExample/ReactExample.tsx]: I'm wondering why this isn't needed anymore?

Comment (thecodejack) [reply src/client/rsg-components/ReactExample/ReactExample.tsx]: All this component is doing is rendering a component with `state`, `setState` which are not required now. So I directly rendered the ExampleComponent here https://github.com/styleguidist/react-styleguidist/blob/6d548d08f7cb111523a6ddd26512df052810689d/src/client/rsg-components/ReactExample/ReactExample.tsx#L47

File: docs/Cookbook.md (hunk 1 above)
Comment (thecodejack) [reply docs/Cookbook.md]: done. Removed

File: docs/Documenting.md (hunk 2 above)
Comment (thecodejack) [reply docs/Documenting.md]: done

File: docs/Documenting.md (hunk 4 above)
Comment (thecodejack) [reply docs/Documenting.md]: done. cleared

**Comment by sapegin** [comment]
Merged the docs!

**Comment by sapegin** [comment]
Hm, wait, there's more than docs ;-/

**Comment by thecodejack** [comment]
my bad...the testing changes... forgot about that..should I create another PR?

**Pull request merged**